pkg/storage/s3/storage.go

```diff
@@ -0,0 +1,107 @@
+package s3
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/gomods/athens/pkg/config"
+)
+
+// Storage implements (github.com/gomods/athens/pkg/storage).Saver and
+// also provides a function to fetch the location of a module
+// Storage uses amazon aws go SDK which expects these env variables
+// - AWS_REGION 			- region for this storage, e.g 'us-west-2'
+// - AWS_ACCESS_KEY_ID		-
+// - AWS_SECRET_ACCESS_KEY 	-
+// - AWS_SESSION_TOKEN		- [optional]
+// For information how to get your keyId and access key turn to official aws docs: https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/setting-up.html
+type Storage struct {
+	bucket  string
+	client  *s3.S3
+	baseURI *url.URL
+}
+
+// New creates a new azure CDN saver
```

Comment:
@michalpristas can you change this Godoc to indicate that it creates a new S3 saver?

Reply:
definitely, many thanks for pointing this out